.github/workflows/ci_cd_main.yml

```diff
@@ -0,0 +1,41 @@
+name: main
+on:
+  push:
+    branches:
+      - main
+
+env:
+  MAIN_PYTHON_VERSION: '3.12'
+  DOCUMENTATION_CNAME: 'actions.docs.ansys.com'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
```

Comment:
This one will need contents: write if I am not mistaken @jorgepiloto -- you need to publish to gh-pages